src/sparseml/pytorch/torchvision/train.py

```diff
@@ -441,21 +443,46 @@ def collate_fn(batch):
         alpha = 1.0 - args.model_ema_decay
         alpha = min(1.0, alpha * adjust)
         model_ema = utils.ExponentialMovingAverage(
-            model_without_ddp, device=device, decay=1.0 - alpha
+            model, device=device, decay=1.0 - alpha
         )
 
-    if args.resume:
-        checkpoint = torch.load(args.resume, map_location="cpu")
-        model_without_ddp.load_state_dict(checkpoint["model"])
-        if not args.test_only:
-            optimizer.load_state_dict(checkpoint["optimizer"])
-            lr_scheduler.load_state_dict(checkpoint["lr_scheduler"])
+    if args.checkpoint_path:
```

Comment:
looks functionally correct for now - but would be nice to break out into a helper function that takes in the checkpoint, resume, and model and returns the loaded model and manager(s)

right now, the overall flow is a bit dense (line 449-481), especially since it happens in two stages managing different requirements. might be good to have just one stage for each (checkpoint, resume, base)